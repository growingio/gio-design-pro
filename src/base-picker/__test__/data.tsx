import React from 'react';
import { Button } from '@gio-design/components';
import { PlusCircleFilled } from '@gio-design/icons';
import List from '../../list';

export const renderItems = () => (
  <>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
  </>
);

export const renderGroups = () => (
  <>
    <List.ItemGroup title="分组 1">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 2">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 3">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
  </>
);

export const footer = (
  <Button type="text" icon={<PlusCircleFilled />} size="small">
    新建内容
  </Button>
);

export const tabNavItems = [
  { key: 'tab-1', children: '选项 1' },
  { key: 'tab-2', children: '选项 2' },
];
