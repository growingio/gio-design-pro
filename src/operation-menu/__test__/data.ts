export const options = [
  {
    disabled: false,
    tooltip: '',
    value: 'update-space-role',
    label: '修改站点角色',
  },
  {
    disabled: false,
    tooltip: '',
    value: 'update-space',
    label: '删除用户',
  },
  {
    disabled: true,
    tooltip: '行行行',
    value: 'update-spxxxace',
    label: '删除用户2',
  },
];

export default {
  options,
};
