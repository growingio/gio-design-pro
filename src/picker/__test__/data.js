export const nodes = [
  {
    label: 'Data 1',
    value: 'data-1',
  },
  {
    label: 'Data 2',
    value: 'data-2',
  },
  {
    label: 'Data 3',
    value: 'data-3',
  },
];

export default {
  nodes,
};
