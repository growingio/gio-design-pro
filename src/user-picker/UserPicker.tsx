import React from 'react';
import { filter, keyBy, includes, uniq, debounce } from 'lodash';
import { Button } from '@gio-design/components';
import { PlusCircleFilled } from '@gio-design/icons';
import BasePicker from '../base-picker';
import SegmentCard from './SegmentCard';
import { useLocalStorage } from '../hooks';
import { UserPickerProps } from './interfaces';
import { resourceToItem } from '../utils';
import { preparedSegments } from './constant';
import { Resource } from '../utils/interfaces';

function UserPicker({
  itemOnHoverDelay = 750,
  segments,
  userId,
  onSelect,
  onCreateSegment,
  updatingRecentDelay = 0,
}: UserPickerProps) {
  const mappedSegements = React.useMemo<{ [key: string]: Resource }>(
    () => keyBy([...preparedSegments, ...segments], 'id'),
    [preparedSegments, segments]
  );
  const [hoveredResource, setHoveredResource] = React.useState<Resource>();
  const [recentSegments, setRecentSegments] = useLocalStorage<string[]>(`${userId}-segments`, []);
  const [query, setQuery] = React.useState<string>('');
  const [scope, setScope] = React.useState<string>('my');
  const [detailVisible, setDetailVisible] = React.useState(false);
  const detailDebounced = debounce(() => {
    setDetailVisible(true);
  }, itemOnHoverDelay);

  function filterSegment(s: Resource) {
    if (query.length > 0) {
      return includes(s.name.toLocaleLowerCase(), query) || includes(s.creator?.toLocaleLowerCase(), query);
    }
    return true;
  }

  function mapResource(current: Resource) {
    const onClick = () => {
      onSelect?.(current.id, current);
      const updateRecent = () => {
        const segIds = uniq([current.id, ...recentSegments]);
        if (segIds.length > 5) {
          setRecentSegments(segIds.slice(0, 5));
        } else {
          setRecentSegments(segIds);
        }
      };
      if (updatingRecentDelay > 0) {
        debounce(updateRecent, updatingRecentDelay)();
      } else {
        updateRecent();
      }
    };
    const onMouseEnter = () => {
      setHoveredResource(current);
      detailDebounced();
    };
    const onMouseLeave = () => {
      setHoveredResource(undefined);
      detailDebounced.cancel();
      setDetailVisible(false);
    };
    return resourceToItem(current, { onClick, onMouseEnter, onMouseLeave });
  }

  const dataSource = React.useMemo(() => {
    if (scope === 'my') {
      const preparedGroup = {
        key: 'prepared',
        title: '预定义',
        items: preparedSegments.map(mapResource),
      };
      const otherGroup = {
        key: 'other',
        title: '其他',
        items: filter(segments, (s) => s.creatorId === userId && filterSegment(s)).map(mapResource),
      };
      if (recentSegments.length > 0) {
        const recentGroup = {
          key: 'recent',
          title: '最近使用',
          items: recentSegments.filter((id) => !!mappedSegements[id]).map((id) => mapResource(mappedSegements[id])),
        };
        return [recentGroup, preparedGroup, otherGroup];
      }
      return [preparedGroup, otherGroup];
    }
    return [...preparedSegments, ...filter(segments, filterSegment)].map(mapResource);
  }, [preparedSegments, segments, recentSegments, scope, query]);

  const items = React.useRef([
    { key: 'my', children: '我的' },
    { key: 'all', children: '全部' },
  ]);
  const footer = (
    <Button type="text" icon={<PlusCircleFilled />} size="small" onClick={onCreateSegment}>
      新建分群
    </Button>
  );
  return (
    <BasePicker
      searchBar={{
        placeholder: '搜索分群名',
        onSearch: setQuery,
      }}
      tabNav={{
        items: items.current,
        onChange: setScope,
      }}
      items={dataSource}
      footer={footer}
      detailVisible={detailVisible}
      renderDetail={() => hoveredResource && <SegmentCard {...hoveredResource} />}
    />
  );
}

export default UserPicker;
