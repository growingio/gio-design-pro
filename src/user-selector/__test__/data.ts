export const currentUserId = 'V0G5vDAR';

export const segment = {
  id: 'y9pm1pme',
  name: '过去 7 天浏览过详情页的地活跃用户',
  detector: {
    usersRatio: 0.002,
    totalUsers: 32,
  },
  creator: '李四',
  creatorId: 'V0G5vDAR',
  createdAt: '2020-05-11T03:45:31.275Z',
  updatedAt: '2020-05-11T03:45:31.275Z',
};

export const segments = [
  {
    id: 'y9pm1pme',
    name: '过去 7 天浏览过详情页的地活跃用户',
    detector: {
      usersRatio: 0.002,
      totalUsers: 32,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-11T03:45:31.275Z',
    updatedAt: '2020-05-11T03:45:31.275Z',
  },
  {
    id: 'AbQ3bDYe',
    name: '过去 7 天低活跃的免费用户',
    detector: {
      usersRatio: 0.0,
      totalUsers: 0,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-11T04:01:53.514Z',
    updatedAt: '2020-05-11T04:01:53.514Z',
  },
  {
    id: 'rRGoYQml',
    name: '过去 7 天访问用户',
    detector: {
      usersRatio: 0.042,
      totalUsers: 832,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-20T08:32:30.387Z',
    updatedAt: '2020-05-20T08:32:30.387Z',
  },
  {
    id: '3KlwQik2',
    name: '过去 7 天登录用户',
    detector: {
      usersRatio: 0.014,
      totalUsers: 247,
    },
    creator: '王武',
    creatorId: 'o8D03G7P',
    createdAt: '2020-05-20T08:32:30.387Z',
    updatedAt: '2020-05-20T08:32:30.387Z',
  },
  {
    id: 'WlGk4Daj',
    name: '未转化用户',
    detector: {
      usersRatio: 0.059,
      totalUsers: 129347,
    },
    creator: '张三',
    creatorId: '7RDYeDA8',
    createdAt: '2020-05-09T02:15:20.643Z',
    updatedAt: '2020-05-09T02:15:20.643Z',
  },
  {
    id: 'klGvyp7E',
    name: '已转化用户',
    detector: {
      usersRatio: 0.034,
      totalUsers: 112379,
    },
    creator: '陈留',
    creatorId: 'nKD1PGWr',
    createdAt: '2020-05-22T02:44:06.972Z',
    updatedAt: '2020-05-22T02:44:06.972Z',
  },
];

export default {
  segment,
};
