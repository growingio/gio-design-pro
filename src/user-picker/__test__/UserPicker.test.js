import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount, render } from 'enzyme';
import UserPicker from '..';
import { currentUserId, preparedSegments, segments } from './data';

describe('UserPicker', () => {
  const onCreateSegment = jest.fn();
  const defaultProps = {
    preparedSegments,
    segments,
    userId: currentUserId,
    onCreateSegment,
  };
  const defaultComponent = <UserPicker {...defaultProps} />;

  it('render my segements', () => {
    const wrapper = mount(defaultComponent);
    wrapper.simulate('click');
    expect(render(wrapper)).toMatchSnapshot();
    const dropdown = document.body.querySelector('.gio-picker-dropdown');
    expect(dropdown).not.toBeNull();
    expect(dropdown).toMatchSnapshot();
  });

  it('render all segments', () => {
    const wrapper = mount(defaultComponent);
    wrapper.simulate('click');
    const dropdown = document.body.querySelector('.gio-picker-dropdown');
    const tabs = dropdown.querySelectorAll('.gio-tabnav-item');
    expect(tabs).toHaveLength(2);
    act(() => {
      tabs[1].click();
    });
    expect(dropdown).toMatchSnapshot();
  });

  it('can select a segment', () => {
    const onSelect = (s) => {
      expect(s).not.toBeNull();
    };
    const wrapper = mount(defaultComponent);
    wrapper.setProps({ onSelect });
    wrapper.simulate('click');
    act(() => {
      document.querySelector('.gio-picker__list .cascader-menu-item').click();
    });
  });
});
