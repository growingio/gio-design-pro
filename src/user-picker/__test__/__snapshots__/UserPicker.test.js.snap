// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPicker render all segments 1`] = `
<div
  class="gio-dropdown gio-picker-dropdown"
  style="opacity: 0; pointer-events: none;"
>
  <div
    class="gio-dropdown-content"
  >
    <div
      class="gio-dropdown-arrow"
    >
      <span
        class="gio-dropdown-arrow-content"
      />
    </div>
    <div
      class="gio-dropdown-inner"
      role="tooltip"
    >
      <div
        class="gio-picker"
      >
        <div
          class="gio-picker__header"
        >
          <div
            class="gio-searchbar"
          >
            <div
              class="gio-input gio-input--middle gio-input__container"
            >
              <input
                class="gio-input__content gio-input__content-suffix"
                placeholder="搜索分群名或创建人"
                type="text"
                value=""
              />
              <div
                class="gio-input__suffix"
              >
                <span
                  class="gio-icon gio-searchbar-suffix-search"
                >
                  <svg
                    class="gio-icon-svg"
                    fill="currentColor"
                    height="16px"
                    viewBox="0 0 14.01 14"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style />
                    </defs>
                    <g>
                      <g>
                        <path
                          d="M6 1a5 5 0 11-5 5 5 5 0 015-5zm4.48 10.19a1 1 0 00-1.28-.13A5.83 5.83 0 016 12a6 6 0 115.07-2.79 1 1 0 00.13 1.26c.93 1 2.65 2.68 2.65 2.68a.49.49 0 01-.7.7z"
                          fill="currentColor"
                          fill-rule="evenodd"
                          id="search-outlined_svg__search"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <button
            class="gio-btn gio-btn-secondary gio-btn-icon-only"
            type="button"
          >
            <span
              class="gio-icon"
            >
              <svg
                class="gio-icon-svg"
                fill="currentColor"
                height="16px"
                viewBox="0 0 14 14"
                width="16px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <style />
                </defs>
                <g>
                  <g>
                    <path
                      d="M6 6.51A.54.54 0 006.51 6V.49a.49.49 0 011 0V6a.54.54 0 00.49.51h5.48a.49.49 0 010 1H8a.54.54 0 00-.51.49v5.48a.49.49 0 01-1 0V8A.54.54 0 006 7.49H.49a.49.49 0 010-1z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      id="plus-outlined_svg__plus"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </button>
        </div>
        <div
          class="gio-tabnav gio-tabnav-line gio-tabnav-sm"
        >
          <div
            class="gio-tabnav-item"
          >
            <div
              class="gio-tabnav-item-btn"
            >
              我的
            </div>
          </div>
          <div
            class="gio-tabnav-item gio-tabnav-item-active"
          >
            <div
              class="gio-tabnav-item-btn"
            >
              全部
            </div>
          </div>
          <div
            class="gio-tabnav-ink-bar"
            style="left: 0px; width: 0px;"
          />
        </div>
        <div
          class="cascader-menu-outer gio-picker__list"
        >
          <div
            class="cascader-menu cascader-menu"
            data-deepsearch="false"
            data-depth="0"
            role="menu"
          >
            <div
              class="cascader-menu-body"
            >
              <div
                class="cascader-menu-group"
              >
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        全部用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        全部用户
                      </h4>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        新用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        新用户
                      </h4>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        活跃用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        活跃用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            310
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            1.60%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        张三
                         | 
                        2020/05/09
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        过去7天浏览过详情页的地活跃用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        过去7天浏览过详情页的地活跃用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            32
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0.20%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        李四
                         | 
                        2020/05/11
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        过去7天低活跃的免费用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        过去7天低活跃的免费用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0.00%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        李四
                         | 
                        2020/05/11
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        过去7天访问用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        过去7天访问用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            832
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            4.20%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        王武
                         | 
                        2020/05/20
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        高意向人群
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        高意向人群
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            64
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0.30%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        陈留
                         | 
                        2020/05/22
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserPicker render my segements 1`] = `
<div
  class="gio-picker-trigger"
  style="cursor:pointer"
>
  <div
    class="gio-input gio-input--middle gio-input__container"
  >
    <input
      aria-label="picker-value"
      class="gio-input__content gio-input__content-suffix"
      placeholder=""
      type="text"
      value=""
    />
    <div
      class="gio-input__suffix"
    >
      <span
        class="gio-icon"
      >
        <svg
          class="gio-icon-svg"
          fill="currentColor"
          height="16px"
          viewBox="0 0 14 14"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <g>
            <g>
              <path
                d="M7.39 9.51a.5.5 0 01-.78 0l-3-3.7a.51.51 0 01-.06-.53A.5.5 0 014 5h6a.5.5 0 01.45.28.51.51 0 01-.06.53z"
                fill="currentColor"
                fill-rule="evenodd"
                id="down-filled_svg__down-2"
              />
            </g>
          </g>
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`UserPicker render my segements 2`] = `
<div
  class="gio-dropdown gio-picker-dropdown"
  style="opacity: 0; pointer-events: none;"
>
  <div
    class="gio-dropdown-content"
  >
    <div
      class="gio-dropdown-arrow"
    >
      <span
        class="gio-dropdown-arrow-content"
      />
    </div>
    <div
      class="gio-dropdown-inner"
      role="tooltip"
    >
      <div
        class="gio-picker"
      >
        <div
          class="gio-picker__header"
        >
          <div
            class="gio-searchbar"
          >
            <div
              class="gio-input gio-input--middle gio-input__container"
            >
              <input
                class="gio-input__content gio-input__content-suffix"
                placeholder="搜索分群名或创建人"
                type="text"
                value=""
              />
              <div
                class="gio-input__suffix"
              >
                <span
                  class="gio-icon gio-searchbar-suffix-search"
                >
                  <svg
                    class="gio-icon-svg"
                    fill="currentColor"
                    height="16px"
                    viewBox="0 0 14.01 14"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style />
                    </defs>
                    <g>
                      <g>
                        <path
                          d="M6 1a5 5 0 11-5 5 5 5 0 015-5zm4.48 10.19a1 1 0 00-1.28-.13A5.83 5.83 0 016 12a6 6 0 115.07-2.79 1 1 0 00.13 1.26c.93 1 2.65 2.68 2.65 2.68a.49.49 0 01-.7.7z"
                          fill="currentColor"
                          fill-rule="evenodd"
                          id="search-outlined_svg__search"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <button
            class="gio-btn gio-btn-secondary gio-btn-icon-only"
            type="button"
          >
            <span
              class="gio-icon"
            >
              <svg
                class="gio-icon-svg"
                fill="currentColor"
                height="16px"
                viewBox="0 0 14 14"
                width="16px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <style />
                </defs>
                <g>
                  <g>
                    <path
                      d="M6 6.51A.54.54 0 006.51 6V.49a.49.49 0 011 0V6a.54.54 0 00.49.51h5.48a.49.49 0 010 1H8a.54.54 0 00-.51.49v5.48a.49.49 0 01-1 0V8A.54.54 0 006 7.49H.49a.49.49 0 010-1z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      id="plus-outlined_svg__plus"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </button>
        </div>
        <div
          class="gio-tabnav gio-tabnav-line gio-tabnav-sm"
        >
          <div
            class="gio-tabnav-item gio-tabnav-item-active"
          >
            <div
              class="gio-tabnav-item-btn"
            >
              我的
            </div>
          </div>
          <div
            class="gio-tabnav-item"
          >
            <div
              class="gio-tabnav-item-btn"
            >
              全部
            </div>
          </div>
          <div
            class="gio-tabnav-ink-bar"
            style="left: 0px; width: 0px;"
          />
        </div>
        <div
          class="cascader-menu-outer gio-picker__list"
        >
          <div
            class="cascader-menu cascader-menu"
            data-deepsearch="false"
            data-depth="0"
            role="menu"
          >
            <div
              class="cascader-menu-body"
            >
              <div
                class="cascader-menu-group"
              >
                <div
                  class="cascader-menu-group-name"
                >
                  预定义
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        全部用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        全部用户
                      </h4>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        新用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        新用户
                      </h4>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="cascader-menu-group"
              >
                <div
                  class="cascader-menu-group-name"
                >
                  其他
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        过去7天浏览过详情页的地活跃用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        过去7天浏览过详情页的地活跃用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            32
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0.20%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        李四
                         | 
                        2020/05/11
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="cascader-menu-item"
                  data-checked="false"
                  data-child-checked="false"
                  role="menuitem"
                >
                  <div
                    aria-haspopup="false"
                    aria-pressed="false"
                    class="cascader-menu-item-inner"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="cascader-menu-item-content"
                    >
                      <div>
                        过去7天低活跃的免费用户
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="gio-picker-hover-panel"
                  >
                    <div
                      class="gio-user-picker-segment"
                    >
                      <h4
                        class="gio-user-picker-segment__name"
                      >
                        过去7天低活跃的免费用户
                      </h4>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        分群人数和占比：
                      </div>
                      <div
                        class="gio-user-picker-segment__tags"
                      >
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          class="gio-tag gio-tag-size-middle gio-tag-type-normal"
                        >
                          <span
                            class="gio-tag-label"
                          >
                            0.00%
                          </span>
                        </span>
                      </div>
                      <div
                        class="gio-user-picker-segment__label"
                      >
                        创建人 & 更新时间：
                      </div>
                      <div
                        class="gio-user-picker-segment__value"
                      >
                        李四
                         | 
                        2020/05/11
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
